src/js/view/utils/resize-listener.js

```diff
@@ -50,44 +67,34 @@ export default class ResizeListener {
         this.expandElement = expandElement;
         this.expandChild = expandChild;
         this.contractElement = contractElement;
-        this.hiddenElement = element.appendChild(resizeElement);
+        this.hiddenElement = resizeElement;
         this.element = element;
-        this.callback = callback;
-        this.resizeRaf = -1;
-        this.lastWidth = this.currentWidth = 0;
-        this.scrollListener = (e) => {
-            cancelAnimationFrame(this.resizeRaf);
-            this.resizeRaf = requestAnimationFrame(() => {
-                const currentWidth = this.currentWidth = element.offsetWidth;
-                if (this.lastWidth === currentWidth) {
-                    return;
-                }
-                this.callback(e, currentWidth);
-            });
-            this.resetTriggers();
-        };
-
-        requestAnimationFrame(() => {
-            this.currentWidth = element.offsetWidth;
-            this.resetTriggers();
-            element.addEventListener('scroll', this.scrollListener, true);
-        });
-    }
-
-    resetTriggers() {
-        const currentWidth = this.currentWidth;
-        this.contractElement.scrollLeft = currentWidth * 2;
-        style(this.expandChild, { width: currentWidth + 1 });
-        this.expandElement.scrollLeft = currentWidth + 1;
-        this.lastWidth = currentWidth;
+        this.view = view;
+        this.model = model;
+        this.width = 0;
+        this.resized = false;
+        if (element.firstChild) {
+            element.insertBefore(resizeElement, element.firstChild);
+        } else {
+            element.appendChild(resizeElement);
+        }
+        element.addEventListener('scroll', scrollListener, true);
+        instances.push(this);
+        scrollListener();
     }
 
     destroy() {
-        if (this.callback) {
+        if (this.view) {
+            const index = instances.indexOf(this);
+            if (index !== -1) {
+                instances.splice(index, 1);
+            }
             this.element.removeEventListener('scroll', this.scrollListener, true);
             this.element.removeChild(this.hiddenElement);
             this.scrollListener =
-                this.callback = null;
+                this.view =
```

Comment:
`this.scrollListener = null`?
`this.view = null`?

Reply:
updated